src/soap/serializers/variable.js

```diff
@@ -38,3 +38,103 @@ export function variableInfoType({ datastream, unitCV, variableCV }) {
       : '')
   )
 }
+
+export function variableDetails({ datastream, refsMap, unitCV }) {
+  const variableElement = (el, key) => {
```

Comment:
So I'm wondering if the number of repetitive get calls for the same key will have a negative performance impact. I know it's probably minor, but it's being done a lot and we may want to consider just setting a bunch of consts. Something like:

```
  const variableElement = (el, key) => {
    const variable = refsMap && refsMap.get(`his.odm.variables.${key}`)
    return variable
      ? `<${el}>${encodeXML(variable)}</${el}>`
      : ''
  }
```

Other examples below.